File: Roemisch/src/test/java/berlin/gd/Roemisch.java
package berlin.gd;

import java.util.Arrays;
import java.util.Optional;

public class Roemisch {

    enum RomanValues {
        C(100), XC(90),
        L(50),  XL(40),
        X(10),  IX(9),
        V(5),   IV(4),
        I(1);
        private int decimalValue;
        RomanValues(int decimalValue) {
            this.decimalValue = decimalValue;
        }
        public int getDecimalValue() {
            return decimalValue;
        }
    }
    public static String toRoman(int num) {
        return Arrays.stream(RomanValues.values())
                .filter(r -> num >= r.getDecimalValue())
                .findFirst()
                .map(r -> r.name() + toRoman(num - r.getDecimalValue()))
                .orElse("");
    }

    public static int toInt(String roman) {
        return Arrays.stream(RomanValues.values())
                .filter(r -> roman.startsWith(r.name()))
                .findFirst()
                .map(r -> r.getDecimalValue() + toInt(roman.substring(r.name().length())))
                .orElse(0)
        ;
    }

}

File: CodingKatas/old/katas/roemisch/history/Roemisch.kt
package katas.roemisch.history

class Roemisch {


    companion object {
        @JvmStatic
        fun toInt(roemisch: String): Int {
            return when {
                roemisch.startsWith("X")  -> 10 + toInt(roemisch.substring(1))
                roemisch.startsWith("IX") -> 9  + toInt(roemisch.substring(2))
                roemisch.startsWith("V")  -> 5  + toInt(roemisch.substring(1))
                roemisch.startsWith("IV") -> 4  + toInt(roemisch.substring(2))
                roemisch.startsWith("I")  -> 1  + toInt(roemisch.substring(1))
                else -> 0
            }
        }

        @JvmStatic
        fun toRoman(num: Int): String {
            return when {
                num >= 10 -> "X"  + toRoman(num - 10)
                num ==  9 -> "IX" + toRoman(0)
                num >=  5 -> "V"  + toRoman(num - 5)
                num ==  4 -> "IV" + toRoman(0)
                num >=  1 -> "I"  + toRoman(num - 1)
                else -> ""
            }
        }
    }




}